plugins/optimization-detective/detection.php

```diff
@@ -38,11 +38,11 @@
  *
  * @global WP_Query $wp_query WordPress Query object.
  *
- * @return int|null Post ID or null if none found.
+ * @return positive-int|null Post ID or null if none found.
  */
 function od_get_cache_purge_post_id(): ?int {
 	$queried_object = get_queried_object();
-	if ( $queried_object instanceof WP_Post ) {
+	if ( $queried_object instanceof WP_Post && $queried_object->ID > 0 ) {
```

Comment:
Is this just to satisfy PHPStan? Seems a bit odd of a check.

Reply:
Yes, it is for the sake of PHPStan. It's so we can say that the function returns a `positive-int`. Technically `$queried_object->ID` can be zero or negative since at the `the_post` action it can be overridden. In fact, the Customizer does this for nav menus which aren't yet saved in the DB: it provides negative values for the IDs to indicate nav menu items that haven't yet been inserted.